assets/scripts/info_bubble/WidthControl.jsx

```diff
@@ -0,0 +1,302 @@
+    const update = () => {
+      const processedValue = processWidthInput(value)
+      if (processedValue) {
+        const normalizedValue = resizeSegment(this.props.segmentEl, RESIZE_TYPE_TYPING, processedValue, false, false)
+        this.props.changeSegmentWidth(this.props.position, normalizedValue)
+      }
+    }
+
+    this.timerId = window.setTimeout(update, WIDTH_EDIT_INPUT_DELAY)
+
+    trackEvent('INTERACTION', 'CHANGE_WIDTH', 'INPUT_FIELD', null, true)
+  }
+
+  onClickInput = (event) => {
+    const el = event.target
+
+    this.setState({
+      isEditing: true,
+      displayValue: undecorateWidth(this.props.value)
+    })
+
+    if (document.activeElement !== el) {
+      el.select()
+    }
+  }
+
+  onDoubleClickInput = (event) => {
+    const el = event.target
+    el.select()
+  }
+
+  onFocusInput = (event) => {
+    const el = event.target
+
+    if (document.activeElement !== el) {
+      el.select()
+    }
+  }
+
+  /**
+   * On blur, input shows prettified width value.
+   */
+  onBlurInput = (event) => {
+    this.setState({
+      isEditing: false,
+      displayValue: prettifyWidth(this.props.value)
+    })
+  }
+
+  /**
+   * Prevent mousedown event from resetting the input view.
+   */
+  onMouseDownInput = (event) => {
+    this.setState({
+      isEditing: true,
+      displayValue: undecorateWidth(this.props.value)
+    })
+  }
+
+  /**
+   * On mouse over, UI assumes user is ready to edit.
+   */
+  onMouseOverInput = (event) => {
+    // Bail if already in editing mode.
+    if (this.state.isEditing) return
+
+    this.setState({
+      displayValue: undecorateWidth(this.props.value)
+    })
+
+    // Automatically select the value on hover so that it's easy to start typing new values.
+    // In React, this only works if the .select() is called at the end of the execution
+    // stack, so we put it inside a setTimeout() with a timeout of zero. We also must
+    // store the reference to the event target because the React synthetic event will
+    // not persist into the setTimeout function.
+    const target = event.target
+    window.setTimeout(() => {
+      target.focus()
+      target.select()
+    }, 0)
+  }
+
+  /**
+   * On mouse out, if user is not editing, UI returns to default view.
+   */
+  onMouseOutInput = (event) => {
+    // Bail if already in editing mode.
+    if (this.state.isEditing) return
+
+    this.setState({
+      displayValue: prettifyWidth(this.props.value)
+    })
+
+    event.target.blur()
+  }
+
+  onKeyDownInput = (event) => {
+    switch (event.keyCode) {
+      case KEYS.ENTER:
+        const normalizedValue = resizeSegment(this.props.segmentEl, RESIZE_TYPE_TYPING, event.target.value, false, false)
+        this.props.changeSegmentWidth(this.props.position, normalizedValue)
+
+        this.setState({
+          isEditing: false
+        })
+
+        this.inputEl.focus()
+        this.inputEl.select()
+
+        break
+      case KEYS.ESC:
```

Comment:
Is this ESC handler working? just not the other one?

Reply:
This isn't working either; I'll duplicate the comment.